src/services/upload/exifService.ts

```diff
@@ -1,90 +1,54 @@
-import EXIF from 'exif-js';
+import exifr from 'exifr';
 
 import { logError } from 'utils/sentry';
 import { NULL_LOCATION, Location } from './metadataService';
 
-const SOUTH_DIRECTION = 'S';
-const WEST_DIRECTION = 'W';
-const CHUNK_SIZE_FOR_EXIF_READING = 4 * 1024 * 1024;
+const EXIF_TAGS_NEEDED = [
+    'DateTimeOriginal',
+    'CreateDate',
+    'ModifyDate',
+    'GPSLatitude',
+    'GPSLongitude',
+    'GPSLatitudeRef',
+    'GPSLongitudeRef',
+];
 interface ParsedEXIFData {
     location: Location;
     creationTime: number;
 }
 
 export async function getExifData(
-    worker,
     receivedFile: globalThis.File
 ): Promise<ParsedEXIFData> {
     try {
-        const fileChunk = await worker.getUint8ArrayView(
-            receivedFile.slice(0, CHUNK_SIZE_FOR_EXIF_READING)
-        );
-        const exifData = EXIF.readFromBinaryFile(fileChunk.buffer);
+        const exifData = await exifr.parse(receivedFile, EXIF_TAGS_NEEDED);
```

Comment:
we can pass the file directly to the package and it will read it using fileReader so custom chunking is not need

Reply:
Can you confirm that  `exifr.parse` won't not fail when any tag in `EXIF_TAGS_NEEDED` is missing?

Reply:
yes have tested that